service/common/src/main/java/org/apache/polaris/service/auth/LocalRSAKeyProvider.java

```diff
@@ -48,12 +76,7 @@ public LocalRSAKeyProvider(Path publicKeyFileLocation, Path privateKeyFileLocati
    */
   @Override
   public PublicKey getPublicKey() {
-    try {
-      return PemUtils.readPublicKeyFromFile(publicKeyFileLocation, "RSA");
-    } catch (IOException e) {
-      LOGGER.error("Unable to read public key from file {}", publicKeyFileLocation, e);
-      throw new RuntimeException("Unable to read public key from file " + publicKeyFileLocation, e);
-    }
+    return publicKey;
```

Comment:
This change will require server restarts for key rotation. I'm not sure if on-the-fly key rotation was envisioned as a supported use case before, but it might be worth flagging this behaviour change for visibility.

Reply:
You'd have to keep the previous (public) key around and referenced by their id, as the [auth0 RSAKeyProvider interface](https://javadoc.io/doc/com.auth0/java-jwt/3.2.0/com/auth0/jwt/interfaces/RSAKeyProvider.html) defines, which [Polaris' KeyProvider](https://github.com/apache/polaris/blob/1631da00b5f1116f779a044b00bbff8132cb4f49/service/common/src/main/java/org/apache/polaris/service/auth/KeyProvider.java) does not. If you change the keys in a running Polaris system, all current tokens become immediately invalid.

Reply:
Implementing the ability to rotate keys consistently across multiple Polaris instances is a bigger effort, and should be backed by a secrets manager.

Reply:
To be clear: I personally support merging this PR.